drivers/counter/counter_mchp_xec.c

```diff
@@ -294,7 +299,11 @@ static int counter_xec_init(const struct device *dev)
 	counter->PRLD = counter_cfg->info.max_top_value;
 	counter->CNT = counter_cfg->info.max_top_value;
 
+#if defined(CONFIG_BOARD_MEC172XEVB_ASSY6906)
```

Comment:
Figure out how to do this so its not board specific

Reply:
It feels like this is a SoC difference and not a board specific one.  If that is really the case, than I suggest just changing the ifdef to be based on SoC.

Reply:
@galak Do you mean replace CONFIG_BOARD_MEC172XEVB_ASSY6906 with CONFIG_SOC_MEC172X_NSZ?
If yes, I have pushed a new patch and rebased. Thanks to review.